.github/workflows/release.yml

```diff
@@ -0,0 +1,103 @@
+name: Publish to crates.io
+
+on:
+  workflow_dispatch:
+    inputs:
+      dry_run:
+        description: 'Dry run mode (create branch but skip tag creation and use cargo publish --dry-run)'
+        required: false
+        type: boolean
+        default: true
```

Comment:
Trivial: I prefer a boolean that is false by default and that we can enable if required. Maybe reverse it to something like do_release or make_release or release_mode which will be false by default.